pkg/loop/telem.go

```diff
@@ -0,0 +1,85 @@
+package loop
+
+import (
+	"context"
+	"github.com/smartcontractkit/chainlink-relay/pkg/loop/internal"
+
+	grpcprom "github.com/grpc-ecosystem/go-grpc-middleware/providers/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+)
+
+type Telemetry = internal.Telemetry
+
+// SetupTelemetry initializes open telemetry and returns a prometheus [http.Handler], client side [grpc.DialOption]s, and a
+// [*grpc.Server] constructor that includes interceptor options.
+func SetupTelemetry(registerer prometheus.Registerer) Telemetry {
```

Comment:
Is this more specific than Telemetry? maybe setup GRPCTelemetry? or SetupInterceptor and return a TelemetryInterceptor?

reading the code in the other pr, from the other repo, the scope of the `Telemetry` was not clear.

Reply:
I had something GRPC specific, but was trying to set this up in a forward compatible way, so that we don't have to break the API as we add things later.

Reply:
So I think this is odd because we are able to use the same type in two different contexts currently (telem and broker), because the only telemetry things are grpc options right now. So the question is whether in the future if/when we return more telemetry things (e.g. i was returning prometheus stuff from here - note the stale comment :facepalm: ), if those things will also be passed in to the `Broker`, or if we will only pass along the grpc options. :thinking:

Reply:
I think what I really wanted was:
```go
type Telemetry struct {
	GRPCOpts GRPCOpts
}
```
And it seemed weird to have a struct with a single field. It's probably too soon to be fretting over API compatibility anyways - I'll reconsider just `type GRPCOpts`... :thinking: